### PR TITLE
Make owner a relationship on Meme

### PR DESCRIPTION
**Pull request by danielpassos**
This PR:

* Simplify the login screen
* Improve login screen constraints
* Update the memeolist schema based on data-sync-server (master)
* Make owner a relationship on Meme
* Display owner info (like the avatar) on meme list
* Put the new meme from subscription on top of the list (to be easily visible there is a new one)
* Fix and improve meme list layout
* Probably something else I can't remember :blush:

Related issues:

* [AEROGEAR-7873](https://issues.jboss.org/browse/AEROGEAR-7873) - Make owner a relationship on Meme

### CODE REVIEW

File: Memeolist/Memeolist/view/list-meme/MemesListViewController.swift [hunk 1]
```diff
@@ -94,7 +92,7 @@ class MemesListViewController: UITableViewController {
         if (segue.identifier == "CommentsSegue") {
             if let button = sender as? UIButton, let cell = button.superview?.superview as? MemesTableViewCell {
                 let destination = segue.destination as? CommentsViewController
-                destination?.meme = memes?.first(where: { $0.id == cell.memeId })
+                destination?.meme = memes.first(where: { ($0?.id)! == cell.memeId! }) as? MemeDetails
```
Comment (danielpassos) [inline Memeolist/Memeolist/view/list-meme/MemesListViewController.swift:95]: @wtrocki I don't know how to make it less ugly :(

**Comment by wtrocki** [comment]
@danielpassos Is that working with server master branch?

**Comment by wtrocki** [comment]
@danielpassos I'm trying to get this tested using community cluster, but schema have changed. Is this working with latest master ?

**Review by wtrocki — APPROVED** [review]

File: Memeolist/Memeolist/view/list-meme/MemesListViewController.swift (hunk 1 above)
Comment (wtrocki) [reply Memeolist/Memeolist/view/list-meme/MemesListViewController.swift:95]: We can assign id as tag to the cell and fetch it.

**Comment by danielpassos** [comment]
Yes, I have tested it locally using the data-server master branch. I did not use the community cluster because I want to also test the subscription and we don't have it there yet.

**Pull request merged**